tabular/src/autogluon/tabular/task/tabular_prediction/predictor.py

```diff
@@ -91,14 +91,43 @@ def __init__(self, learner):
         self._learner: Learner = learner  # Learner object
         self._learner.persist_trainer(low_memory=True)
         self._trainer: AbstractTrainer = self._learner.load_trainer()  # Trainer object
-        self.output_directory = self._learner.path
-        self.problem_type = self._learner.problem_type
-        self.eval_metric = self._learner.eval_metric
-        self.label_column = self._learner.label
-        self.feature_metadata = self._trainer.feature_metadata
-        self.class_labels = self._learner.class_labels
-        self.class_labels_internal = self._learner.label_cleaner.ordered_class_labels_transformed
-        self.class_labels_internal_map = self._learner.label_cleaner.inv_map
+
+    @property
+    def class_labels(self):
+        return self._learner.class_labels
+
+    @property
+    def class_labels_internal(self):
+        return self._learner.label_cleaner.ordered_class_labels_transformed
+
+    @property
+    def class_labels_internal_map(self):
+        return self._learner.label_cleaner.inv_map
+
+    @property
+    def eval_metric(self):
+        return self._learner.eval_metric
+
+    @property
+    def problem_type(self):
+        return self._learner.problem_type
+
+    @property
+    def feature_metadata(self):
+        return self._trainer.feature_metadata
+
+    @property
+    def feature_metadata_in(self):
+        return self._learner.feature_generator.feature_metadata_in
+
+    # TODO: v0.1 rename to label
+    @property
+    def label_column(self):
+        return self._learner.label
```

Comment:
nit: This implies single label classification. Do we expect to add multi-label classification support at some point?

Reply:
I think we may consider to support multi-label prediction. I'm adding https://www.kaggle.com/c/google-quest-challenge to the multimodal benchmark and it has 30 labels.

Reply:
Yes, label_column will be changed to label, this is only to support legacy API during transition process.